### PR TITLE
Config with zope product template

### PR DESCRIPTION
**Pull request by icemac**
- Drop support for Python 2.7, 3.5, 3.6.
- Drop ``zserver`` extra in ``setup.py``. Thus dropping FTP support.
- Drop support for ``Zope < 5``.

### CODE REVIEW

File: src/Products/PluggableAuthService/Extensions/upgrade.py [hunk 1]
```diff
@@ -155,7 +155,7 @@ def descend(user_folder, obj):
                         new_map[npid] = map[key]
                         _write(RESPONSE,
                                'upgradeLocalRoleAssignmentsFromRoot',
-                               '  Translated %s to %s\n' % (key, npid))
+                               '  Translated {} to {}\n'.format(key, npid))
```
Comment (dataflake) [inline src/Products/PluggableAuthService/Extensions/upgrade.py]:
```suggestion
                               f'  Translated {key} to {npid}\n')
```

File: src/Products/PluggableAuthService/PluggableAuthService.py [hunk 2]
```diff
@@ -610,8 +609,8 @@ def _extractUserIds(self, request, plugins):
 
                         except _SWALLOWABLE_PLUGIN_EXCEPTIONS:
                             reraise(auth)
-                            msg = 'AuthenticationPlugin %s error' % (
-                                authenticator_id,)
+                            msg = 'AuthenticationPlugin {} error'.format(
+                                authenticator_id)
```
Comment (dataflake) [inline src/Products/PluggableAuthService/PluggableAuthService.py]:
```suggestion
                            msg = f'AuthenticationPlugin {authenticator_id} error'
```

File: src/Products/PluggableAuthService/plugins/DomainAuthHelper.py [hunk 3]
```diff
@@ -274,7 +274,7 @@ def manage_addMapping(self, user_id='', match_type='', match_string='',
         except re.error:
             msg = 'Invalid regular expression %s' % match_string
         except ValueError as e:
-            msg = 'Invalid match string %s (%s)' % (match_string, e)
+            msg = 'Invalid match string {} ({})'.format(match_string, e)
```
Comment (dataflake) [inline src/Products/PluggableAuthService/plugins/DomainAuthHelper.py]:
```suggestion
            msg = f'Invalid match string {match_string} ({e})'
```

File: src/Products/PluggableAuthService/plugins/ZODBGroupManager.py [hunk 4]
```diff
@@ -117,10 +116,10 @@ def enumerateGroups(self, id=None, title=None, exact_match=False,
                 info.update(self._groups[group_id])
 
                 info['pluginid'] = plugin_id
-                info['properties_url'] = '%s?%s' % (e_url, p_qs)
-                info['members_url'] = '%s?%s' % (e_url, m_qs)
+                info['properties_url'] = '{}?{}'.format(e_url, p_qs)
+                info['members_url'] = '{}?{}'.format(e_url, m_qs)
```
Comment (dataflake) [inline src/Products/PluggableAuthService/plugins/ZODBGroupManager.py]:
```suggestion
                info['properties_url'] = f'{e_url}?{p_qs}'
                info['members_url'] = f'{e_url}?{m_qs}'
```

File: src/Products/PluggableAuthService/plugins/ZODBRoleManager.py [hunk 5]
```diff
@@ -145,8 +145,8 @@ def enumerateRoles(self, id=None, exact_match=False, sort_by=None,
                 info.update(self._roles[role_id])
 
                 info['pluginid'] = plugin_id
-                info['properties_url'] = '%s?%s' % (e_url, p_qs)
-                info['members_url'] = '%s?%s' % (e_url, m_qs)
+                info['properties_url'] = '{}?{}'.format(e_url, p_qs)
+                info['members_url'] = '{}?{}'.format(e_url, m_qs)
```
Comment (dataflake) [inline src/Products/PluggableAuthService/plugins/ZODBRoleManager.py]:
```suggestion
                info['properties_url'] = f'{e_url}?{p_qs}'
                info['members_url'] = f'{e_url}?{m_qs}'
```

File: src/Products/PluggableAuthService/plugins/ZODBUserManager.py [hunk 6]
```diff
@@ -209,7 +202,7 @@ def enumerateUsers(self, id=None, login=None, exact_match=False,
                 info = {'id': self.prefix + user_id,
                         'login': self._userid_to_login[user_id],
                         'pluginid': plugin_id,
-                        'editurl': '%s?%s' % (e_url, qs)}
+                        'editurl': '{}?{}'.format(e_url, qs)}
```
Comment (dataflake) [inline src/Products/PluggableAuthService/plugins/ZODBUserManager.py]:
```suggestion
                        'editurl': f'{e_url}?{qs}'}
```

File: src/Products/PluggableAuthService/plugins/tests/test_ChallengeProtocolChooser.py [hunk 7]
```diff
@@ -147,7 +142,7 @@ def setup_user(self):
         self.assertIsNone(pas.getUserById('test_user_'))
 
         username, password = 'test_user_', 'test_user_pw'
-        self.basic_auth = '%s:%s' % (username, password)
+        self.basic_auth = '{}:{}'.format(username, password)
```
Comment (dataflake) [inline src/Products/PluggableAuthService/plugins/tests/test_ChallengeProtocolChooser.py]:
```suggestion
        self.basic_auth = f'{username}:{password}'
```

File: src/Products/PluggableAuthService/tests/test_PluggableAuthService.py [hunk 8]
```diff
@@ -341,7 +339,7 @@ def __repr__(self):
 
     def publishable(self, *args, **kw):
 
-        return 'Args: %s\nKeywords: %s' % (args, kw)
+        return 'Args: {}\nKeywords: {}'.format(args, kw)
```
Comment (dataflake) [inline src/Products/PluggableAuthService/tests/test_PluggableAuthService.py]:
```suggestion
        return f'Args: {args}\nKeywords: {kw}'
```

**Review by dataflake — APPROVED** [review]
Just some f-strings

**Comment by icemac** [comment]
Thank you for reviewing this PR. 😃

**Pull request merged**